rpcserver.go

```diff
@@ -510,36 +504,45 @@ func (r *rpcServer) MintAsset(ctx context.Context,
 		return nil, err
 	}
 
-	decDisplay := req.Asset.DecimalDisplay
-	metaType := req.Asset.AssetMeta.Type
+	var seedlingMeta *proof.MetaReveal
+	if req.Asset.AssetMeta != nil {
+		// Ensure that the meta type is valid.
+		metaType, err := proof.IsValidMetaType(req.Asset.AssetMeta.Type)
+		if err != nil {
+			return nil, err
+		}
 
-	// If decimal display is specified by the user, but the meta type is not
-	// JSON, then we have to error out as the value needs to be encoded in
-	// the meta field. If an opaque meta field is specified, then we simply
-	// don't know how to handle it.
-	if decDisplay != 0 && metaType != taprpc.AssetMetaType_META_TYPE_JSON {
-		return nil, fmt.Errorf("decimal display requires JSON meta " +
-			"type")
-	}
+		// If the meta type is not JSON, then a custom decimal display
+		// cannot be set.
+		if metaType != proof.MetaJson && req.Asset.DecimalDisplay != 0 {
+			return nil, fmt.Errorf("cannot set decimal display " +
+				"if meta type is not JSON")
+		}
 
-	// Check if decimal display exceeds max allowed value.
-	if decDisplay > maxDecDisplay {
-		return nil, fmt.Errorf("decimal display cannot exceed %v",
-			maxDecDisplay)
-	}
+		// If the asset meta field was specified, then the data inside
+		// must be valid. Let's check that now.
+		seedlingMeta = &proof.MetaReveal{
+			Data: req.Asset.AssetMeta.Data,
+			Type: metaType,
+		}
 
-	// If a JSON meta field is specified, then we'll update the decimal
-	// display in the meta field. Even if the decimal display is not
-	// specified in the rpc request, we'll still encode a 0 value.
-	if req.Asset.AssetMeta.Type == taprpc.AssetMetaType_META_TYPE_JSON {
-		updatedMeta, err := taprpc.EncodeDecimalDisplayInJSON(
-			decDisplay, req.Asset.AssetMeta.Data,
-		)
+		err = seedlingMeta.Validate()
 		if err != nil {
 			return nil, err
 		}
 
-		req.Asset.AssetMeta.Data = updatedMeta
+		// If a custom decimal display was requested, add that to the
+		// metadata and re-validate it.
+		if metaType == proof.MetaJson && req.Asset.DecimalDisplay != 0 {
+			updatedMeta, err := seedlingMeta.SetDecDisplay(
+				req.Asset.DecimalDisplay,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			seedlingMeta = updatedMeta
+		}
```

Comment:
I think the comment promises that we execute `seedlingMeta.Validate()` again, but I don't think we do.

Reply:
I've added the re-validate bit that the in-code comment called for.